pallets/parachain-staking/src/lib.rs

```diff
@@ -936,11 +873,6 @@ pub mod pallet {
 		}
 	}
 
-	#[pallet::storage]
-	#[pallet::getter(fn delay_nomination_exits_migration)]
-	/// True if executed, false by default
-	type DelayNominationExitsMigration<T: Config> = StorageValue<_, bool, ValueQuery>;
```

Comment:
We're removing the type from code, but the data will still be stored in the on-chain storage. A single bool will definitely not bloat state in any practical way. Still I wonder if we should clean it up.

If so we could either add the cleanup logic to `on_runtime_upgrade` which would then have to be removed again later. Or we could do it manually with a set_storage extrinsic.

Reply:
Nice catch, I think removing it in `on_runtime_upgrade` is simple and makes sense.

Reply:
I added back the storage item, but killed whatever it stored. In a future PR after this runtime upgrade is run, I'll remove the storage item itself.

I'm a bit confused to be honest. Does `kill()` mean that it's still storing `bool::default() == false` in storage? Any suggestions here?

Maybe the extrinsic approach is better...

Reply:
kill() means it is not storing anything in storage. Since you declared it as a `ValueQuery` however, the runtime and tools like polkadot js will _act_ as if it stored `bool::default()`. This should be fine though because the migration code is gone so nothing should be reading it.

Reply:
So we need 2 upgrades before removing the on_runtime_upgrade right ?

Reply:
Yes, one more after this one